tests/test_models.py

```diff
@@ -814,27 +814,31 @@ def test_user_email_property_user_missing(self):
         enterprise_customer_user = factories.EnterpriseCustomerUserFactory(user_id=42)
         assert enterprise_customer_user.user_email is None
 
-    def test_on_create_should_set_user_fk_to_user_id(self):
+    def test_on_create_should_set_user_fk_to_user(self):
         """Test that user_fk is set to user_id when creating a record."""
-        user_id = 123
-        enterprise_customer_user = factories.EnterpriseCustomerUserFactory(user_id=user_id)
-        enterprise_customer_user.refresh_from_db()
+        user = factories.UserFactory(email='email@example.com')
+        enterprise_customer_user = factories.EnterpriseCustomerUserFactory(user_id=user.id)
 
-        assert enterprise_customer_user.user_fk == user_id, (
-            f"Expected user_fk to be {user_id}, but got {enterprise_customer_user.user_fk}"
+        assert enterprise_customer_user.user_fk == user, (
+            f"Expected user_fk to be User with id {user.id}, but got {enterprise_customer_user.user_fk}"
         )
 
-    def test_on_update_should_set_user_fk_to_user_id(self):
+    def test_on_update_should_set_user_fk_to_user(self):
         """Test that user_fk updates when user_id is modified and saved."""
-        enterprise_customer_user = factories.EnterpriseCustomerUserFactory(user_id=100)
+        user = factories.UserFactory(email='email@example.com')
+        second_user = factories.UserFactory(email='email2@example.com')
+        enterprise_customer_user = factories.EnterpriseCustomerUserFactory(user_id=user.id)
+
+        assert enterprise_customer_user.user_fk == user, (
+            f"Expected user_fk to be User with id {user.id}, but got {enterprise_customer_user.user_fk}"
+        )
```

Comment:
it might be good to do a couple Django ORM queries in here to test that the FK relationship is "groked" by the ORM, for example
```python
assert EnterpriseCustomerUser.objects.filter(user_fk=user).count() == 1
assert user.enterprise_customer_users.all().count() == 1
```